BaseStation/Controller Controls/Program.cs

```diff
@@ -0,0 +1,68 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading;
+using System.Threading.Tasks;
+using OpenTK.Input;
+using Scarlet.Communications;
+using Scarlet.Utilities;
+
+// this code will be used for the basestation and takes in all of the controller inputs then sends
+// them to the rover side. Whichever is client or server side has yet to be determined
```

Comment:
The architecture of Scarlet is set up such that the base station should be a server and the various components of the rover shall be clients.

Reply:
Will this cause issue for communication between rover components? If the jetson needs to talk to the drive system, then it would have to do two wireless hops to communicate.

Reply:
I've been discussing this with Jaden, and this would be essentially a complete re-write of our networking systems, adding a lot more complexity.

Is this something that is _needed_? Can the controls be architected such that the double-wireless hop is OK?

Alternatively, you may consider having a central server on the rover (like the Jetson), then having all the other components as well as base be a client. Then this double hop would stay on the wired network on the rover. All commands to other parts of the rover would need to be routed through the Jetson though.

Reply:
Putting the server on the rover is the approach I thought we were using to avoid this exact problem. Two wireless hops adds too much latency and too much packet loss when it can be avoided by moving the server.

Alternatively, we could adopt a message passing/pub sub system, and then the broker could live anywhere we want.

Reply:
If you'd like, putting the server on the rover as an intermediate step between the controller and other rover parts would work fine.

The reason we were not recommending that by default is that if anything goes wrong with the server, the whole rover loses functionality. With the base being a server and all over components connecting to it like we originally intended, rover systems can individually drop without complete loss of functionality.

Architecture-wise, there should be no problem doing it your way. You'll just need to make sure that the server is very well coded for reliability to avoid these issues.

Reply:
Running the server on a separate process with an OS level watchdog such as upstart should provide that reliability.